src/sentry/mediators/external_requests/util.py

```diff
@@ -73,17 +80,16 @@ def send_and_save_sentry_app_request(url, sentry_app, org_id, event, **kwargs):
         # Response code of 0 represents timeout
         buffer.add_request(response_code=0, org_id=org_id, event=event, url=url)
         # Re-raise the exception because some of these tasks might retry on the exception
-        raise
-
-    else:
-        track_response_code(resp.status_code, slug, event)
-        buffer.add_request(
-            response_code=resp.status_code,
-            org_id=org_id,
-            event=event,
-            url=url,
-            error_id=resp.headers.get("Sentry-Hook-Error"),
-            project_id=resp.headers.get("Sentry-Hook-Project"),
-        )
-        resp.raise_for_status()
-        return resp
+        raise e
```

Comment:
Any specific reason we want to make this change? Just `raise` is usually preferred so that we have the original stacktrace unmodified.

Reply:
Oops I didn't know it re-raised. I thought it was raising a new, empty `RuntimeError`.